src/NuGetGallery.Core/GitHub/GitHubUsageConfiguration.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace NuGetGallery.GitHub
+{
+    public class GitHubUsageConfiguration : IGitHubUsageConfiguration
+    {
+        public GitHubUsageConfiguration(IReadOnlyList<RepositoryInformation> reposCache)
+        {
+            NuGetPackagesGitHubDependencies = reposCache.Any()
+                ? GitHubCacheTransformer.GetNuGetPackagesDependents(reposCache)
+                : new Dictionary<string, NuGetPackageGitHubInformation>();
+        }
+
+        public Dictionary<string, NuGetPackageGitHubInformation> NuGetPackagesGitHubDependencies { get; }
+
+        public NuGetPackageGitHubInformation GetPackageInformation(string packageId)
+        {
+            return NuGetPackagesGitHubDependencies.ContainsKey(packageId)
```

Comment:
I would use `TryGetValue` here to avoid two look ups.

Also, you allocate each time a `packageId` isn't found. You can avoid this by creating a private static readonly field that contains an "empty" `NuGetPackageGitHubInformation`, and returning that instead of create a new object.